marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/artifacts/clearUserArtifacts.sjs

```diff
@@ -0,0 +1,33 @@
+/**
+ Copyright (c) 2020 MarkLogic Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+'use strict';
+//No specific privilege available for writing step definition or legacy mappings
+xdmp.securityAssert("http://marklogic.com/data-hub/privileges/write-ingestion", "execute");
+xdmp.securityAssert("http://marklogic.com/data-hub/privileges/write-mapping", "execute");
+xdmp.securityAssert("http://marklogic.com/data-hub/privileges/write-match-merge", "execute");
+xdmp.securityAssert("http://marklogic.com/data-hub/privileges/write-custom", "execute");
+xdmp.securityAssert("http://marklogic.com/data-hub/privileges/write-flow", "execute");
+xdmp.securityAssert("http://marklogic.com/data-hub/privileges/write-entity-model", "execute");
+
+const consts = require('/data-hub/5/impl/consts.sjs');
+const DataHubSingleton = require('/data-hub/5/datahub-singleton.sjs');
+const dataHub = DataHubSingleton.instance();
+
+let urisToDelete = cts.uris(null, null, cts.andQuery([cts.collectionQuery(consts.USER_ARTIFACT_COLLECTIONS.concat('http://marklogic.com/data-hub/mappings'))
+  ,cts.notQuery(cts.collectionQuery(consts.HUB_ARTIFACT_COLLECTION))])).toArray();
+let database = xdmp.databaseName(xdmp.database());
+urisToDelete.forEach(uri => dataHub.hubUtils.deleteDocument(uri, database));
```

Comment:
No need to do an xdmp.invoke here - just do xdmp.documentDelete. We know we're not trying to do this in some other database.